### DISCUSSION TITLE
Live migrations

### DISCUSSION THREAD

**Discussion by nikitashamgunov** [firecracker-microvm/firecracker#3119] [Ideas]
Team,

Are you planning to implement live migrations?

**Comment by xmarcalx** [comment]
Hi,

I think we should clarify what you intend with live migration first.
Firecracker was designed to mainly support server less ephemeral workloads (which usually have a duration of ms up to 3s more or less).  For such workloads, "live migration" is not really a thing because you can simply wait for that workload to finish and you do not need to live migrate it incurring a cost that can be orders of magnitude bigger than the task itself.

Nevertheless, if for live migration we intend the capability to move a microVM from one host to another, then Firecracker already support it 😃 . Thanks to the microVMs fast startup and the capability to take a snapshot and resume a microVM from it in a matter of ms, Firecracker does provide a simple implementation of what live migration is usually used for!
